i6_models/primitives/mixup.py

```diff
@@ -0,0 +1,153 @@
+__all__ = ["MixupConfig", "Mixup"]
+
+import random
+from dataclasses import dataclass
+
+import torch
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class MixupConfig(ModelConfiguration):
+    """
+    Attributes:
+        buffer_size: number of frames.
+        apply_prob: probability to apply mixup at all
+        lambda_min: minimum lambda value
+        lambda_max: maximum lambda value
+        max_num_mix: maximum number of mixups (random int in [1, max_num_mix])
+    """
+
+    buffer_size: int = 1_000_000
+    apply_prob: float = 1.0
+    max_num_mix: int = 4
+    lambda_min: float = 0.003
+    lambda_max: float = 0.3
+
+
+class FeatureBuffer(torch.nn.Module):
+    """
+    The FeatureBuffer saves the feature from previous timeframes
+    after the buffer is full, the buffer will not be updated
+    """
+
+    def __init__(self, *, buffer_size: int, feature_dim: int):
+        self.filled = False
+        self.pos = 0
+        self.buffer_size = buffer_size
+        self.cache = torch.nn.parameter.Parameter(
+            data=torch.zeros((self.buffer_size, feature_dim)), requires_grad=False
+        )
+
+    def append(self, tensor: torch.Tensor):
+        t_dim = tensor.shape[0]
+        if t_dim > self.buffer_size:
+            tensor = tensor[: self.buffer_size]
+            t_dim = self.buffer_size
+        delta_pos = min(self.buffer_size - self.pos, t_dim)
+        end_pos = self.pos + delta_pos
+        self.cache[self.pos : end_pos] = tensor[:delta_pos]
+        self.pos = end_pos
+
+        if end_pos == self.buffer_size:
+            self.filled = True
+            end_pos = t_dim - delta_pos
+            self.cache[:end_pos] = tensor[delta_pos:]
+            self.pos = end_pos
+
+    def get_random(self, b_dim: int, t_dim: int, max_num_mixup: int, n_mask: torch.tensor) -> torch.Tensor:
```

Comment:
Can you document the shapes (of `n_mask` and returned tensor)?

Also, wrong type:
```suggestion
    def get_random(self, b_dim: int, t_dim: int, max_num_mixup: int, n_mask: torch.Tensor) -> torch.Tensor:
```